client/concordclient/include/client/concordclient/concord_client.hpp

```diff
@@ -108,6 +108,13 @@ struct SubscribeConfig {
   std::string pem_private_key;
 };
 
+struct StateSnapshotConfig {
+  // Number of threads available for state snapshot
+  uint32_t num_threads;
+  // Timeout till grpc connection with replicas will wait.
+  uint16_t timeout_in_sec;
```

Comment:
It might make sense to have default values defined here for the members. We can do it in a separate PR.

Reply:
thanks @dartdart26 , I will take this in a separate PR.